app/touch-bar-menu.js

```diff
@@ -31,14 +31,35 @@ module.exports = function touchBarMenu(window, soundcloud) {
     icon: `${__dirname}/res/like.png`,
     click: () => {
       soundcloud.likeUnlike()
+      toggleLikeUnlikeIcon()
+    }
+  })
+
+  function toggleLikeUnlikeIcon() {
+    if(likeUnlike.icon == `${__dirname}/res/like.png`) {
+      likeUnlike.icon = `${__dirname}/res/liked.png`
+    } else {
+      likeUnlike.icon = `${__dirname}/res/like.png`
+    }
+  }
+
+  const repost = new TouchBarButton({
+    icon: `${__dirname}/res/repost.png`,
+    click: () => {
+      soundcloud.repost()
     }
   })
 
   const trackInfo = new TouchBarLabel()
 
-  soundcloud.on('play', ({ title, subtitle }) => {
+  soundcloud.on('play', ({ title, subtitle, trackMetadata }) => {
     playPause.icon = `${__dirname}/res/pause.png`
     trackInfo.label = formatTitle(title, subtitle)
+    if(trackMetadata.isLiked) {
```

Comment:
The liked/not liked state management is fundamentally flawed here. The liked state can be changed in many other ways than hitting the like button on TouchBar: users can click (several different) like buttons on SoundCloud itself, use keyboard shortcut, menus, etc. This means the TouchBar like state will only be consistent if the user exclusively uses the TouchBar controls, which is unlikely.

What I would suggest is trying to figure out if we can trigger 'liked' events from SoundCloud (preload.js) independent from what triggered the state change and then implement the button state management in a similar fashion to play/pause management here (eg. `soundcloud.on('liked')`). One possible means of change detection would be using [DOM mutation observers](https://developer.mozilla.org/en-US/docs/Web/API/MutationObserver).

I know this sounds complicated but I also think it's not worth merging a half-assed implementation.

Do you think you can figure this out?